java/fury-core/src/main/java/org/apache/fury/builder/MetaSharedCodecBuilder.java

```diff
@@ -190,11 +192,49 @@ protected Expression createRecord(SortedMap<Integer, Expression> recordComponent
   @Override
   protected Expression setFieldValue(Expression bean, Descriptor descriptor, Expression value) {
     if (descriptor.getField() == null) {
-      // Field doesn't exist in current class, skip set this field value.
-      // Note that the field value shouldn't be an inlined value, otherwise field value read may
-      // be ignored.
-      // Add an ignored call here to make expression type to void.
-      return new Expression.StaticInvoke(ExceptionUtils.class, "ignore", value);
+      FieldMismatchCallback.FieldAdjustment adjustment =
+          fury.getFieldMismatchCallback()
+              .onMismatch(beanClass, descriptor.getTypeName(), descriptor.getName());
+
+      if (adjustment == null) {
+        // Field doesn't exist in current class, skip set this field value.
+        // Note that the field value shouldn't be an inlined value, otherwise field value read may
+        // be ignored.
+        // Add an ignored call here to make expression type to void.
+        return new Expression.StaticInvoke(ExceptionUtils.class, "ignore", value);
+      } else {
+
+        Field newTargetField = adjustment.getTargetField();
+
+        // Field doesn't exist in current class, invoke field mismatch callback.
+        Expression fieldMismatchCallback =
+            Expression.Invoke.inlineInvoke(
+                new Expression.Reference(FURY_NAME, TypeRef.of(Fury.class)),
+                "getFieldMismatchCallback",
+                TypeRef.of(FieldMismatchCallback.class),
+                /* needTryCatch */ false);
```

Comment:
This can be a private field, instead of being instantiated every time. I'm not used to this kind of metaprogramming so I need to figure out how to do this.